aloscene/scene_flow.py

```diff
@@ -0,0 +1,127 @@
+        cls,
+        optical_flow: Flow,
+        depth: Depth,
+        next_depth: Depth,
+        intrinsic: CameraIntrinsic,
+    ):
+        """Create scene flow from optical flow, depth a T, depth at T + 1 and the intrinsic
+
+        Parameters
+        ----------
+        optical flow: aloscene.Flow
+            The optical flow at T.
+        depth: aloscene.Depth
+            The depth at T."
+        next_depth: aloscene.Depth
+            The depth at T + 1
+        intrinsic : aloscene.CameraIntrinsic
+            The intrinsic of the image at T.
+        """
+        start_vector = depth.as_points3d(camera_intrinsic=intrinsic).cpu().numpy()
+        end_vector = next_depth.as_points3d(camera_intrinsic=intrinsic).cpu().numpy()
+
+        scene_flow_vector = end_vector - start_vector
+        scene_flow_vector = np.reshape(scene_flow_vector, (3, depth.H, depth.W))
+
+        masked_points = Mask(np.isfinite(scene_flow_vector).all(0), names=("H", "W"))
+
+        result = torch.from_numpy(scene_flow_vector)
+        tensor = cls(result)
+        tensor.append_mask(masked_points)
+        tensor.append_occlusion(optical_flow.occlusion.clone(), "occlusion")
+        return tensor
+
+    def append_occlusion(self, occlusion: Mask, name: Union[str, None] = None):
+        """Attach an occlusion mask to the scene flow.
+
+        Parameters
+        ----------
+        occlusion: aloscene.Mask
+            Occlusion mask to attach to the Scene Flow
+        name: str
+            If none, the occlusion mask will be attached without name (if possible). Otherwise if no other unnamed
+            occlusion mask are attached to the scene flow, the mask will be added to the set of mask.
+        """
+        self._append_child("occlusion", occlusion, name)
+
+    def _hflip(self, **kwargs):
+        """Flip scene flow horizontally.
+
+        Returns
+        -------
+        flipped_scene_flow : aloscene.SceneFlow
+            horizontally flipped scene flow map
+        """
+        flow_flipped = super()._hflip(**kwargs)
+        # invert x axis of flow vector
+        labels = flow_flipped.drop_children()
+        sl_x = flow_flipped.get_slices({"C": 0})
+        sl_z = flow_flipped.get_slices({"C": 2})
+        flow_flipped[sl_x] = -1 * flow_flipped[sl_x]
+        flow_flipped[sl_z] = -1 * flow_flipped[sl_z]
+        flow_flipped.set_children(labels)
+        return flow_flipped
+
+    def _vflip(self, **kwargs):
+        """Flip scene flow vertically.
+
+        Returns
+        -------
+        flipped_scene_flow : aloscene.SceneFlow
+            vertically flipped scene flow map
+        """
+        flow_flipped = super()._vflip(**kwargs)
+        # invert y axis of flow vector
+        labels = flow_flipped.drop_children()
+        sl_y = flow_flipped.get_slices({"C": 1})
+        sl_z = flow_flipped.get_slices({"C": 2})
+        flow_flipped[sl_y] = -1 * flow_flipped[sl_y]
+        flow_flipped[sl_z] = -1 * flow_flipped[sl_z]
```

Comment:
Same comment as for hflip. The Z coordinate should not change during vertical flip.

Reply:
It"s fix.